xarray/backends/zarr.py

```diff
@@ -126,14 +126,19 @@ def _determine_zarr_chunks(enc_chunks, var_chunks, ndim):
     # threads
     if var_chunks and enc_chunks_tuple:
         for zchunk, dchunks in zip(enc_chunks_tuple, var_chunks):
-            for dchunk in dchunks:
+            for dchunk in dchunks[:-1]:
                 if dchunk % zchunk:
                     raise NotImplementedError(
                         "Specified zarr chunks %r would overlap multiple dask "
                         "chunks %r. This is not implemented in xarray yet. "
                         " Consider rechunking the data using "
                         "`chunk()` or specifying different chunks in encoding."
                         % (enc_chunks_tuple, var_chunks))
+            if dchunks[-1] > zchunk:
+                raise ValueError(
+                    "Final chunk of Zarr array must be smaller than first. "
+                    "Variable %r has incompatible chunks. Consider rechunking "
+                    "using `chunk()`." % var_chunks)
```

Comment:
In this error, `%r` is referring to `var_chunks`, not the variable name. I would output both `enc_chunks_tuple` and `var_chunks`, as this information is useful for debugging.

Reply:
@lilyminium - if you can address this comment, we can get your PR merged.